test/gateway/eip20_cogateway/confirm_revert_stake_intent.js

```diff
@@ -0,0 +1,308 @@
+
+    let blockHeight = new BN(StubData.gateway.revert_stake.proof_data.block_number, 16);
+    let storageRoot = StubData.gateway.revert_stake.proof_data.storageHash;
+
+    let storageProof = StubData.gateway.revert_stake.proof_data.storageProof[0].serializedProof;
+    await coGateway.setStorageRoot(blockHeight, storageRoot);
+
+    let tx = await coGateway.confirmRevertStakeIntent(
+      messageHash, blockHeight, storageProof
+    );
+
+    assert.strictEqual(
+      tx.receipt.status,
+      true,
+      'Transaction should success.',
+    );
+
+  });
+
+  it('should emit RevertStakeIntentConfirmed event', async function () {
+
+    await coGateway.setInboxStatus(messageHash, MessageStatusEnum.Declared);
+
+    let blockHeight = new BN(StubData.gateway.revert_stake.proof_data.block_number, 16);
+    let storageRoot = StubData.gateway.revert_stake.proof_data.storageHash;
+
+    let storageProof = StubData.gateway.revert_stake.proof_data.storageProof[0].serializedProof;
+    await coGateway.setStorageRoot(blockHeight, storageRoot);
+
+    let tx = await coGateway.confirmRevertStakeIntent(
+      messageHash, blockHeight, storageProof
+    );
+
+    assert.strictEqual(
+      tx.receipt.status,
+      true,
+      'Transaction should success.',
+    );
+
+    let event = EventDecoder.getEvents(tx, coGateway);
+
+    assert.isDefined(
+      event.RevertStakeIntentConfirmed,
+      'Event `RevertStakeIntentConfirmed` must be emitted.',
+    );
+
+    let eventData = event.RevertStakeIntentConfirmed;
+    assert.strictEqual(
+      eventData._messageHash,
+      messageHash,
+      `Expected message hash ${messageHash} is different from message hash from 
+      event ${eventData._messageHash}`,
+    );
+
+    assert.strictEqual(
+      eventData._staker,
+      stakeRequest.staker,
+      `Expected staker ${stakeRequest.staker} is different from staker from 
+      event ${eventData._staker}`,
+    );
+
+    let expectedNonce = new BN(stakeRequest.nonce, 16);
+    assert.strictEqual(
+      eventData._stakerNonce.eq(expectedNonce),
+      true,
+      `Expected stakerNonce ${expectedNonce.toNumber()} is different from nonce from 
+      event ${eventData._stakerNonce.toNumber()}`,
+    );
+
+    assert.strictEqual(
+      eventData._amount.eq(amount),
+      true,
+      `Expected amount ${amount.toNumber()} is different from amount from 
+      event ${eventData._amount.toNumber()}`,
+    );
+
+  });
+
+  it('should fail if revert stake intent is already confirmed', async function () {
+
+    await coGateway.setInboxStatus(messageHash, MessageStatusEnum.Declared);
+
+    let blockHeight = new BN(StubData.gateway.revert_stake.proof_data.block_number, 16);
+    let storageRoot = StubData.gateway.revert_stake.proof_data.storageHash;
+
+    let storageProof = StubData.gateway.revert_stake.proof_data.storageProof[0].serializedProof;
+    await coGateway.setStorageRoot(blockHeight, storageRoot);
+
+    await coGateway.confirmRevertStakeIntent(
```

Comment:
either 
```
await coGateway.confirmRevertStakeIntent(messageHash, blockHeight, storageProof); 
```
or 
```
await coGateway.confirmRevertStakeIntent(
    messageHash, 
    blockHeight, 
    storageProof
); 
```

Reply:
Similarly at other location in the test.

Reply:
Actually ;) (note comma last argument)
```js
await coGateway.confirmRevertStakeIntent(
    messageHash, 
    blockHeight, 
    storageProof,
); 
```

Reply:
It's for the same reasons of clarity and readability. Why should it be different for JS?

Reply:
I don't think it had less clarity and readability previously. 🤔 
Though I have changed it. But we should definitely agree on JS standards.